src/components/radio/radio.spec.ts

```diff
@@ -1,370 +1,398 @@
-      builder = tcb;
+describe('MdRadio', () => {
+  let builder: TestComponentBuilder;
+  let dispatcher: MdRadioDispatcher;
+
+  beforeEachProviders(() => [
+    provide(MdRadioDispatcher, {useFactory: () => {
+      dispatcher = new MdRadioDispatcher();
+      return dispatcher;
+    }})
+  ]);
+
+  beforeEach(inject([TestComponentBuilder], (tcb: TestComponentBuilder) => {
+    builder = tcb;
+  }));
+
+  describe('inside of a group', () => {
+    let fixture: ComponentFixture<RadiosInsideRadioGroup>;
+    let groupDebugElement: DebugElement;
+    let groupNativeElement: HTMLElement;
+    let radioDebugElements: DebugElement[];
+    let radioNativeElements: HTMLElement[];
+    let groupInstance: MdRadioGroup;
+    let radioInstances: MdRadioButton[];
+    let testComponent: RadiosInsideRadioGroup;
+
+    beforeEach(async(() => {
+      builder.createAsync(RadiosInsideRadioGroup).then(f => {
+        fixture = f;
+        fixture.detectChanges();
+
+        testComponent = fixture.debugElement.componentInstance;
+
+        groupDebugElement = fixture.debugElement.query(By.directive(MdRadioGroup));
+        groupNativeElement = groupDebugElement.nativeElement;
+        groupInstance = groupDebugElement.injector.get(MdRadioGroup);
+
+        radioDebugElements = fixture.debugElement.queryAll(By.directive(MdRadioButton));
+        radioNativeElements = radioDebugElements.map(debugEl => debugEl.nativeElement);
+        radioInstances = radioDebugElements.map(debugEl => debugEl.componentInstance);
+      });
     }));
 
-    it('should have same name as radio group', (done: () => void) => {
-      builder
-        .overrideTemplate(TestApp, `
-            <md-radio-group name="my_group">
-              <md-radio-button></md-radio-button>
-            </md-radio-group>`)
-        .createAsync(TestApp)
-        .then(fixture => {
-          let button = fixture.debugElement.query(By.css('md-radio-button'));
-
-          fixture.detectChanges();
-          expect(button.componentInstance.name).toBe('my_group');
-        }).then(done);
+    it('should set individual radio names based on the group name', () => {
+      expect(groupInstance.name).toBeTruthy();
+      for (let radio of radioInstances) {
+        expect(radio.name).toBe(groupInstance.name);
+      }
     });
 
-    it('should not allow click selection if disabled', (done: () => void) => {
-      builder
-        .overrideTemplate(TestApp, '<md-radio-button disabled></md-radio-button>')
-        .createAsync(TestApp)
-        .then(fixture => {
-          let button = fixture.debugElement.query(By.css('md-radio-button'));
+    it('should disable click interaction when the group is disabled', () => {
+      testComponent.isGroupDisabled = true;
+      fixture.detectChanges();
 
-          fixture.detectChanges();
-          expect(button.componentInstance.checked).toBe(false);
+      radioNativeElements[0].click();
```

Comment:
We could actually remove the click handler altogether for Radio Buttons as we did with Checkbox; however, if that happens, this test would have to be rewritten to work in IE11.